specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -3517,6 +3517,464 @@
+          "RestorePointCollections"
+        ],
+        "operationId": "RestorePointCollections_ListAll",
+        "description": "Gets the list of restore point collections in the subscription. Use nextLink property in the response to get the next page of restore point collections. Do this till nextLink is not null to fetch all the restore point collections.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RestorePointCollectionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{restorePointName}": {
+      "put": {
+        "operationId": "RestorePoints_Create",
+        "description": "The operation to create the restore point. Updating properties of an existing restore point is not allowed",
+        "x-ms-examples": {
+          "Create a restore point": {
+            "$ref": "./examples/CreateARestorePoint.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "restorePointCollectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the restore point collection."
+          },
+          {
+            "name": "restorePointName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the restore point."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RestorePointCreate"
+            },
+            "description": "Parameters supplied to the Create restore point operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/RestorePoint"
```

Comment:
Do we support update scenarios for existing Restore point ?

Reply:
we do not. from PutVMRestorePointOperationBase.cs
protected void UpdateRestorePoint(DM.VMRestorePoint dmRestorePoint, RequestValidator validator)
        {
            Tracer.Trace(0, "RestorePoint {0} already exists", VMRestorePointName);
            if (!IsModelSame(dmRestorePoint))
            {
                Tracer.Trace(0, "Update of an existing Restore Point is not supported.");
                throw RPApiErrors.OperationNotAllowed(validator.Request, Strings.OperationNotSupported(ApiNames.RestorePoint.UpdateOperation, dmRestorePoint.Name));
            }

Reply:
RestorePoint is proxy only ARM resource, so it also does not have tags. so there are no tags to update

Reply:
Thank you for handling idempotent puts in your service btw 👍

---
In reply to: [592893950](https://github.com/Azure/azure-rest-api-specs/pull/13219#discussion_r592893950) [](ancestors = 592893950)